src/test/java/org/opentripplanner/middleware/bugsnag/BugsnagTest.java

```diff
@@ -1,67 +1,299 @@
+     * event request replaces the previous expired one.
+     */
     @Test
-    public void canCreateBugsnagEvent() {
-        Persistence.bugsnagEvents.create(BUGSNAG_EVENT);
-        BugsnagEvent retrieved = Persistence.bugsnagEvents.getById(BUGSNAG_EVENT.id);
-        assertEquals(BUGSNAG_EVENT.id, retrieved.id, "Found Bugsnag event.");
+    public void processExpiredEventRequest() {
+        // Must match the project id value in eventDataResponse.json
+        String projectId = "79gg84750000d122";
+        // Must match the id value in expiredEventDataRequest.json
+        String expiredEventDataRequestId = "611a9123b6e99d4fcec2dc5c";
+        // Must match the id value in replacedEventDataRequest.json
+        String replacedEventDataRequestId = "5948fhf734h362";
+        BugsnagEventRequest bugsnagEventRequest =
+            createBugsnagEventRequest(
+                projectId,
+                expiredEventDataRequestId,
+                "pending",
+                getDateInPast(2)
+            );
+        // create wiremock stub for the refresh event data request.
+        wireMockServer.stubFor(
+            get(urlMatching(String.format("/projects/%s/event_data_requests/%s", projectId, expiredEventDataRequestId)))
+                .willReturn(
+                    aResponse()
+                        .withBodyFile("expiredEventDataRequest.json")
+                )
+        );
+        // create wiremock stub for new event data request
+        wireMockServer.stubFor(
+            post(urlMatching(String.format("/projects/%s/event_data_requests", projectId)))
+                .willReturn(
+                    aResponse()
+                        .withBodyFile("replacedEventDataRequest.json")
+                )
+        );
+        BugsnagEventHandlingJob bugsnagEventHandlingJob = new BugsnagEventHandlingJob();
+        bugsnagEventHandlingJob.refreshEventRequest(bugsnagEventRequest);
+        BugsnagEventRequest removedBugsnagEventRequest =
+            Persistence.bugsnagEventRequests.getOneFiltered(
+                Filters.and(
+                    Filters.eq("projectId", projectId),
+                    Filters.eq("eventDataRequestId", expiredEventDataRequestId)
+                )
+            );
+        assertNull(removedBugsnagEventRequest);
 
+        BugsnagEventRequest newBugsnagEventRequest =
+            Persistence.bugsnagEventRequests.getOneFiltered(
+                Filters.and(
+                    Filters.eq("projectId", projectId),
+                    Filters.eq("eventDataRequestId", replacedEventDataRequestId)
+                )
+            );
+        assertNotNull(newBugsnagEventRequest);
+        // Rounding hour to nearest whole day bumps this by a day from the original 2 to 3 days.
+        assertEquals(3, newBugsnagEventRequest.daysInPast);
+        assertEquals("PREPARING", newBugsnagEventRequest.status);
+        newBugsnagEventRequest.delete();
```

Comment:
This seems like some kind of post-test cleanup. It might make sense to put this in a `@BeforeEach`-decorated method and always remove all bugsnagEvents/bugsnagEventRequests. This way if a single test fails before hitting the cleanup section, it can be ensured that the database will be good to go. This way tests can be certain to not have effects on other tests.

Same comment at the end in each of the next 6 tests.